addon/-private/relationships/has-many.js

```diff
@@ -1,43 +1,11 @@
-import { get } from '@ember/object';
-import LiveQuery from '../live-query';
-
-export default LiveQuery.extend({
-  _store: null,
-  _model: null,
-  _relationship: null,
-
-  init(...args) {
-    const store = get(this, '_store');
-    const model = get(this, '_model');
-    const relationship = get(this, '_relationship');
-
-    this._sourceCache = store.cache._sourceCache;
-    this._identityMap = store.cache._identityMap;
-    this._query = store.source.queryBuilder.findRelatedRecords(
-      model,
-      relationship
-    );
-
-    this._super(...args);
-  },
+import ReadOnlyArrayProxy from '../system/read-only-array-proxy';
 
+export default ReadOnlyArrayProxy.extend({
   pushObject(record) {
-    const store = get(this, '_store');
-    const model = get(this, '_model');
-    const relationship = get(this, '_relationship');
-
-    return store.update(t =>
-      t.addToRelatedRecords(model.identity, relationship, record.identity)
-    );
+    return this.addToContent(record);
```

Comment:
How are mutations conveyed back to the underlying orbit sources?

Reply:
`addToContent` is passed in on instantiation from the `model`. The problem I had is that by returning a promise from `pushObjects` we are breaking `Ember`'s `ArrayProxy` type. It is one of the reasons I decided to keep `ArrayProxy` in javascript.